state/settings.go

```diff
@@ -266,6 +266,29 @@ func (s *Settings) applyChanges(changes settings.ItemChanges) {
 	}
 }
 
+// WriteBulkSettings writes all of the input settings documents
+// in a single transaction.
+// NOTE: The settings are passed as values and not references.
+// Unlike the call to "Write" on a single settings reference,
+// we cannot expect all the settings objects to be updated to represent
+// consistency between the live and on-disk values.
+// TODO (manadart 2019-04-30): This is inelegant.
+// The settings abstractions are poor, but model generations really makes
+// this logic creak.
+func (st *State) WriteBulkSettings(s []Settings) error {
+	if len(s) == 0 {
+		return nil
+	}
+
+	var allOps []txn.Op
+	for _, cfg := range s {
+		_, ops := cfg.settingsUpdateOps()
+		allOps = append(allOps, ops...)
+	}
+
+	return errors.Annotate(st.db().RunTransaction(allOps), "writing multiple settings")
```

Comment:
This should be turned into a 

st.db.Run(func(attempt int) {
  // do the cfg.settingsUpdateOps here
})

That allows for cases where the changes have already been applied by a different controller, etc.